### PR TITLE
Create NextJS launcher

### PR DESCRIPTION
**Pull request by lubien**

### CODE REVIEW

File: internal/sourcecode/templates/nextjs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+WORKDIR /app
+COPY . .
+RUN yarn install --frozen-lockfile
+
+# If using npm with a `package-lock.json` comment out above and use below instead
+# RUN npm ci
+
+ENV NEXT_TELEMETRY_DISABLED 1
+
+RUN yarn build
```
Comment (lubien) [inline internal/sourcecode/templates/nextjs/Dockerfile:14]: This needs a little bit more work. I realize that `NUXT_PUBLIC` variables need to be set as args so I plan on adding instructions here and on the docs

Comment (jsierles) [reply internal/sourcecode/templates/nextjs/Dockerfile:14]: You can add build args at launch time to `fly.toml` as is [done with Rails](https://github.com/superfly/flyctl/blob/master/internal/sourcecode/scan.go#L206-L210).

File: internal/sourcecode/templates/nextjs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+FROM node:16-alpine AS runner
+WORKDIR /app
+
+ENV NODE_ENV production
+ENV NEXT_TELEMETRY_DISABLED 1
+
+RUN addgroup --system --gid 1001 nodejs
+RUN adduser --system --uid 1001 nextjs
+
+COPY --from=builder /app ./
```
[the remark this answers is not in the document]
Comment (jsierles) [reply internal/sourcecode/templates/nextjs/Dockerfile:34]: I'm assuming this isn't required anymore?

**Review by jsierles — APPROVED** [review]
LGTM, just left a few comments.

File: internal/sourcecode/templates/nextjs/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Install dependencies only when needed
+FROM node:16-alpine AS builder
+# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
+COPY . .
+RUN yarn install --frozen-lockfile
```
[the remark this answers is not in the document]
Comment (lubien) [reply internal/sourcecode/templates/nextjs/Dockerfile:7]: That sounds nice. There are more places we'd need to put it through 🤔

**Pull request merged**